plugins/github-pull-requests-board/src/api/useOctokitGraphQl.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Octokit } from '@octokit/rest';
+import { useApi, githubAuthApiRef } from '@backstage/core-plugin-api';
+
+let octokit: any;
+
+export const useOctokitGraphQl = <T>() => {
+  const auth = useApi(githubAuthApiRef);
+
+  return (path: string, options?: any): Promise<T> =>
+    auth.getAccessToken(['repo'])
+      .then((token: string) => {
+        if(!octokit) {
+          octokit = new Octokit({ auth: token })
```

Comment:
Here I suggest to add support for overwriting the `baseUrl` in the octokit cli, because backstage supports self hosted github so there could be cases where this cli needs to call the internal github.
https://backstage.io/docs/integrations/github/locations#configuration
You should be able to get the configured `apiBaseUrl`. with the `configApiRef`

```suggestion
          octokit = new Octokit({ auth: token, ...(baseUrl && { baseUrl })})
```

Reply:
I just added and tested on DAZN's environment and seems everything is fine.... but as I couldn't find much examples can you give this one a special attention on reviewing? 😅😅😅😅
Like: did I did what was supposed to do? 😅